src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Columns/Content/badge.html.twig

```diff
@@ -25,11 +25,23 @@
 
 <div class="text-right">
 {% if record[column.options.field] is not empty %}
-  <span class="badge rounded badge-{{ column.options.badge_type }}">
-    {{ record[column.options.field] }}
-  </span>
+  {% if record.status is defined %}
```

Comment:
Is it still necessary to complicate this generic template, since there is already a dedicated one for `cart_status_data.html.twig`
Checking `record.status` in a generic template doesn't seem normal

Reply:
Could be possible to reuse _cart_status_data.html.twig_ by creating _Column_ Class for it and also would need to add one more if for green button with one more regexp maybe.

Reply:
The new column might not be necessary, there is an overriding feature in the grid templates Based on grid id, field type, name, ... So the template only could be enough (maybe it just needs to be correctly renamed)

Reply:
I think me and @zuk3975  did not like that override and already created the column on the other PR. This override by name is such a hidden mystery that a new developer might took couple of hours to understand what and why is happening by debugging the code. You expect that your chosen template will be rendered and then the magic happens and another template is rendered (that 'if' in the php file is really hidden behavior and not a beautiful one).

Reply:
I agree with @tdavidsonas88 . Override based on file names is unreliable magic code ("convention over configuration") 🤔

Reply:
Yep I agree with the weird magic, although I understand the idea
I guess it was meant to avoid creating a new column type each time a (small) modification was needed in a column

Reply:
I also agree with Tadas.
At first he have done it accidentally as you say  (without the custom template), we weren't aware of that hidden fallback, so we tried to figure out why the hell its working :smile: It was painful and took some time, so we decided, we just create a new one, so it would be easier to other devs in similar situation.